### PR TITLE
Update to go 1.12.8

### PR DESCRIPTION
**Pull request by cblecker**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Update to go 1.12.8

**Which issue(s) this PR fixes**:
ref #79912

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Update to use go 1.12.8
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by cblecker** [comment]
Still to be done:
- Create and push image
- Update rules_go (https://github.com/bazelbuild/rules_go/issues/2177)

**Comment by cblecker** [comment]
/assign @javier-b-perez 
Would you mind building and pushing the image, Javier?

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Comment by BenTheElder** [comment]
I built / pushed an image, the CL is pending

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -53,7 +53,7 @@ load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_depe
 go_rules_dependencies()
 
 go_register_toolchains(
-    go_version = "1.12.7",
+    go_version = "1.12.8",
```
Comment (BenTheElder) [inline build/root/WORKSPACE]: /lgtm cancel
rules_go fails on this as an unknown version

**Comment by BenTheElder** [comment]
/lgtm cancel
https://github.com/kubernetes/kubernetes/pull/81390#discussion_r313665589 <-- this is going to fail until this is resolved, please don't retest etc. until then

**Comment by cblecker** [comment]
Pushed up a new commit to fix a test failure:

Tests under go1.12.7:
```
$ docker run --rm -v ${PWD}:/go/src/k8s.io/kubernetes -w /go/src/k8s.io/kubernetes golang:1.12.7 go test -v ./vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/... -run '(TestWithInvalidCaCertPath|TestInvalidCaCert)'
=== RUN   TestWithInvalidCaCertPath
--- PASS: TestWithInvalidCaCertPath (0.00s)
=== RUN   TestInvalidCaCert
--- PASS: TestInvalidCaCert (0.00s)
PASS
ok      k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib    0.027s
?       k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/diskmanagers       [no test files]
?       k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures   [no test files]
```

Tests under go1.12.8:
```
docker run --rm -v ${PWD}:/go/src/k8s.io/kubernetes -w /go/src/k8s.io/kubernetes golang:1.12.8 go test -v ./vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/... -run '(TestWithInvalidCaCertPath|TestInvalidCaCert)'
=== RUN   TestWithInvalidCaCertPath
E0814 04:43:25.865720    1179 connection.go:179] Failed to parse URL: <nil>. err: parse https://should-not-matter:should-not-matter: invalid port ":should-not-matter" after host
--- FAIL: TestWithInvalidCaCertPath (0.00s)
    connection_test.go:168: Expected an os.PathError, got: 'parse https://should-not-matter:should-not-matter: invalid port ":should-not-matter" after host' (&url.Error{Op:"parse", URL:"https://should-not-matter:should-not-matter", Err:(*errors.errorString)(0xc00003f8d0)})
=== RUN   TestInvalidCaCert
E0814 04:43:25.866151    1179 connection.go:179] Failed to parse URL: <nil>. err: parse https://should-not-matter:should-not-matter: invalid port ":should-not-matter" after host
--- FAIL: TestInvalidCaCert (0.00s)
    connection_test.go:182: Expected invalid certificate error, got 'parse https://should-not-matter:should-not-matter: invalid port ":should-not-matter" after host'
FAIL
FAIL    k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib    0.024s
?       k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/diskmanagers       [no test files]
?       k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures   [no test files]
```

Tests under go1.12.8 after fix:
```
$ docker run --rm -v ${PWD}:/go/src/k8s.io/kubernetes -w /go/src/k8s.io/kubernetes golang:1.12.8 go test -v ./vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/... -run '(TestWithInvalidCaCertPath|TestInvalidCaCert)'
=== RUN   TestWithInvalidCaCertPath
--- PASS: TestWithInvalidCaCertPath (0.00s)
=== RUN   TestInvalidCaCert
--- PASS: TestInvalidCaCert (0.00s)
PASS
ok      k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib    0.026s
?       k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/diskmanagers       [no test files]
?       k8s.io/kubernetes/vendor/k8s.io/legacy-cloud-providers/vsphere/vclib/fixtures   [no test files]
```

File: staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/connection_test.go [hunk 2]
```diff
@@ -159,7 +159,7 @@ func TestWithValidThumbprint(t *testing.T) {
 func TestWithInvalidCaCertPath(t *testing.T) {
 	connection := &vclib.VSphereConnection{
 		Hostname: "should-not-matter",
-		Port:     "should-not-matter",
```
Comment (imkin) [inline staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/connection_test.go]: This could have been a fixture too may be. But I agree its part of "doesn't" matter. Lets fix this for not.

File: hack/boilerplate/boilerplate_test.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 import boilerplate
 import unittest
-import StringIO
+from io import StringIO
```
Comment (BenTheElder) [inline hack/boilerplate/boilerplate_test.py]: seems unrelated? how did the go bump switch this to python3?

Comment (cblecker) [reply hack/boilerplate/boilerplate_test.py]: Bazel was failing this test locally on my side when I bumped things. May be my setup?

Comment (BenTheElder) [reply hack/boilerplate/boilerplate_test.py]: ah. I thought we had it configured to force python2 /shrug
as long as it passes in CI, seems good to me
